connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -167,4 +169,26 @@ message Expression {
     // (Optional) Alias metadata expressed as a JSON map.
     optional string metadata = 3;
   }
+
+  message LambdaFunction {
+    // (Required) The lambda function.
+    Expression function = 1;
+
+    // (Required) Function arguments. Must not be empty.
+    //
+    // All arguments must be UnresolvedNamedLambdaVariables.
+    repeated Expression arguments = 2;
+
+    // (Required) A lambda function can be hidden when a user wants to
+    // process an completely independent expression in a 'HigherOrderFunction',
+    // the lambda function and its variables are then only used for internal
+    // bookkeeping within the higher order function.
+    bool hidden = 3;
+  }
+
+  message UnresolvedNamedLambdaVariable {
```

Comment:
Can we just make this an unresolved attribute? And do the heavy lifting in the connect planner?

Reply:
Unless there is special resolution in Catalyst on this type otherwise +1 to re-use `unresolved attribute`

Reply:
it are literally the argument names in a lambda, for example `x => x + 1`, the `UnresolvedNamedLambdaVariable` is `x`.